File: app/src/main/java/com/skydoves/waterdays/utils/BundleUtils.java
package com.skydoves.waterdays.utils;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

/**
 * Developed by skydoves on 2017-08-19.
 * Copyright (c) 2017 skydoves rights reserved.
 */

public final class BundleUtils {
    private BundleUtils() {}

    public static Bundle maybeGetBundle(final @Nullable Bundle state, final @NonNull String key) {
        if (state == null) {
            return null;
        }

        return state.getBundle(key);
    }
}

File: app/src/main/java/com/skydoves/waterdays/utils/BundleUtils.kt
package com.skydoves.waterdays.utils

import android.os.Bundle

/**
 * Developed by skydoves on 2017-08-19.
 * Copyright (c) 2017 skydoves rights reserved.
 */

object BundleUtils {
  fun maybeGetBundle(state: Bundle?, key: String): Bundle? {
    state.let { return state?.getBundle(key) }
  }
}
